### PR TITLE
New contest btn

### PR DESCRIPTION
**Pull request by Changron**
1. make new contest button same as problem
2. add warning message before deleting contest
3. fix pass rate in contest detail page
4. add color of is_homework and can_register
5. remove register as user button
6. show freeze time in contest
7. show contest id in contest detail
8. show no submission when no submission
9. add button in editContest 
10. 
    2.
    
    3.
    
    4.
    
    5.
    
    6.
    
    7.
    
    8.
    
    9.

### CODE REVIEW

File: contest/contest_info.py [hunk 1]
```diff
@@ -116,6 +116,10 @@ def get_scoreboard(contest):
         new_contestant.testcases_solved = new_contestant.get_testcases_solved()
         scoreboard.add_user(new_contestant)
 
+    for problem in scoreboard.problems:
+        problem.pass_rate = float(problem.pass_user)/len(scoreboard.users) * 100
```
Comment (bruce3557) [inline contest/contest_info.py]: ```float(problem.pass_user)/len(scoreboard.users) * 100``` -> ```float(problem.pass_user) / len(scoreboard.users) * 100```

File: contest/templates/contest/register.html [hunk 2]
```diff
@@ -11,28 +11,11 @@ <h3 class="modal-title">
...
-      </td>
-    </tr>
-    {% endif %}
-    <tr class="separate-area"></tr>
     {% if groups %}
     <tr>
       <td>
         <a class="btn btn-success btn-lg btn-block" data-toggle="collapse"
-           data-target="#group_select">a Group</a>
+           data-target="#group_select">Group</a>
```
Comment (bruce3557) [inline contest/templates/contest/register.html]: not sure we have convention in JS / HTML?

**Comment by bruce3557** [comment]
Seems not bad in first glance, I will review again later

File: contest/templates/contest/register.html [hunk 3]
```diff
@@ -68,7 +51,7 @@ <h3 class="modal-title">
       <form class="form-inline" method="POST" 
             action="{% url 'contest:register' contest.id %}">
         {% csrf_token %}
-        <label>How many Public User?</label>
+        <label>How many Public User do you want?</label>
```
Comment (nickelcheng) [inline contest/templates/contest/register.html]: How many Public User do you need?

**Comment by nickelcheng** [comment]
I got a ZeroDivisionError
screenshot from 2015-04-24 16 51 01

**Comment by nickelcheng** [comment]
BTW,
1. Can you show tooltip for both pass rate and not pass rate (just like what problem does)
2. Add contest name before add a new contest (just like what problem does)

**Comment by nickelcheng** [comment]
3. Show cid in contest detail page

**Comment by Changron** [comment]
why the contest name first?

**Comment by Changron** [comment]
fixed
btw @bruce3557  what do you mean?

**Comment by nickelcheng** [comment]
Add contest name first for behave same with problem page.
Problem name added first for adding tags and test cases more convenient.
Btw, please add did in contest detail page

**Comment by nickelcheng** [comment]
image

**Comment by nickelcheng** [comment]
1. Can you add cid in contest detail page
2. Can you add a cancel button in contest edit page (which can return to previous page)
3. Can you show No Submission when a problem doesn't have any submission during contest (just like what [problem list](http://140.114.77.129:8080/problem/?page=151) did)

**Comment by nickelcheng** [comment]
Bruce means that he'll review again later

**Comment by Changron** [comment]
what does "not sure we have convention in JS / HTML?" mean?

**Comment by Changron** [comment]
added

**Comment by nickelcheng** [comment]
The redirect button is great!
+1, @bruce3557 please check

**Comment by bruce3557** [comment]
I just read code and find the style of JS/HTML naming and code is different with different people's PR.
So just want to check whether there is a common convention in our development guildline

File: contest/views.py [hunk 4]
```diff
@@ -133,7 +133,7 @@ def new(request):
                 new_contest = form.save()
                 logger.info('Contest: User %s Create a new contest %s!' %
                     (request.user ,new_contest.id))
-                return redirect('contest:archive')
+                return redirect('contest:contest',new_contest.id)
```
Comment (bruce3557) [inline contest/views.py]: ```return redirect('contest:contest',new_contest.id)``` -> ```return redirect('contest:contest', new_contest.id)```

File: contest/views.py [hunk 5]
```diff
@@ -154,17 +154,17 @@ def edit(request, cid):
...
-                    {'form':form,'title':title})
+                    {'form':form, 'title':title, 'cid':contest.id})
         if request.method == 'POST':
             form = ContestForm(request.POST, instance = contest, initial={'method':request.method})
             if form.is_valid():
                 modified_contest = form.save()
                 logger.info('Contest: User %s edited contest %s!' %
                     (request.user, modified_contest.id))
-                return redirect('contest:archive')
+                return redirect('contest:contest',modified_contest.id)
```
Comment (bruce3557) [inline contest/views.py]: Ditto

**Comment by bruce3557** [comment]
+1 after comments fixed
thanks!

**Comment by nickelcheng** [comment]
About JS/HTML naming, there seems no rules for our development QQ

**Comment by Changron** [comment]
fixed

**Comment by nickelcheng** [comment]
going to merge

**Pull request merged**